packages/react_rearch/lib/src/widgets/capsule_warm_up.dart

```diff
@@ -0,0 +1,32 @@
+// part of '../widgets.dart';
+
+// /// Provides [toWarmUpWidget], a mechanism to create a [Widget] from a [List]
+// /// of the current states of some "warm up" [Capsule]s.
+// extension CapsuleWarmUp<T> on List<AsyncValue<T>> {
+//   /// Creates a [Widget] from a [List] of the current states of
+//   /// some "warm up" [Capsule]s.
+//   ///
+//   /// - [child] is returned when all of the current states are [AsyncData].
+//   /// - [loading] is returned when any of the current states are [AsyncLoading].
+//   /// - [errorBuilder] is called to build the returned [Widget] when any
+//   /// of the current states are [AsyncError].
+//   Widget toWarmUpWidget({
+//     required Widget Function(List<AsyncError<T>>) errorBuilder,
+//     required Widget loading,
+//     required Widget child,
+//   }) {
+//     // Check for any errors first
+//     final asyncErrors = whereType<AsyncError<T>>();
+//     if (asyncErrors.isNotEmpty) {
+//       return errorBuilder(asyncErrors.toList());
+//     }
+
+//     // Check to see if we have any still loading
+//     if (any((value) => value is AsyncLoading<T>)) {
+//       return loading;
+//     }
+
+//     // We have only AsyncData (no loading or error), so return the child
+//     return child;
+//   }
+// }
```

Comment:
Wondering if we should add a `warmUpCapsules` field or similar to the top level `RearchBootstrapper` component (or whatever we call it) over in `react_rearch` instead of this extension method, as it predates the ability to do `[capsule1, capsule2].map(use)`, as that was added in a later version.

Reply:
Sure, If I manage to make work the Bootstrap, then the next step would be to replicate flutter's warmup (or improve it)